feat(identity): authenticate a principal with key auth (#5638)
Kong/developer.konghq.com: app/_includes/plugins/key-auth/consumer-keys.md

> When you create [a Consumer](/gateway/entities/consumer/#set-up-a-consumer) or a Principal [LINK] {% new_in 3.15 %}, you can specify a `key`.

Author: ```suggestion
When you create [a Consumer](/gateway/entities/consumer/#set-up-a-consumer) or a [Principal](/identity/principals/#configure-a-principal) {% new_in 3.15 %}, you can specify a `key`.
```

> When you create [a Consumer](/gateway/entities/consumer/#set-up-a-consumer) or a [Principal](/identity/principals/#configure-a-principal) {% new_in 3.15 %}, you can specify a `key`.

Reviewer: ```suggestion
When you create [a Consumer](/gateway/entities/consumer/#set-up-a-consumer) or a [Principal identity](/identity/principals/#configure-a-principal) {% new_in 3.15 %}, you can specify a `key`.
```
Since you technically have to create the principal first then add the credentials

{% navtabs "set a key" %}
{% navtab "Consumers" %}

- Declarative configuration with `keyauth_credentials`
- The `/consumers/{usernameOrId}/{{include.slug}}` endpoint.

{% endnavtab %}
{% navtab "Principals" %}

- A POST request to the `/v2/directories/{directoryId}/principals/{principalId}/api-keys` endpoint

> - The Konnect UI

Reviewer: ```suggestion
- The {{site.konnect_short_name}} UI
```

{% endnavtab %}
{% endnavtabs %}

> When authenticating, Consumers and Principals must specify their key in either the query, body, or header:

Reviewer: ```suggestion
When authenticating, Consumers and principals must specify their key in either the query, body, or header:
```
